examples/networking/misc/lwip_ipv4/main.c

```diff
@@ -0,0 +1,210 @@
+
+    if ((result = sock_udp_create(&sock, NULL, &remote, SOCK_FLAGS_REUSE_EP)) != 0) {
+        printf("sock_udp_create() failed with %d\n", result);
+        return -1;
+    }
+
+    if (data_len != (size_t)sock_udp_send(&sock, argv[3], data_len, NULL)) {
+        printf("sock_udp_send() failed with %d\n", (int)data_len);
+        goto out;
+    }
+
+    if ((result = sock_udp_recv(&sock, buffer, sizeof(buffer) - 1,
+                                CONFIG_CLIENT_TIMEOUT_SEC * US_PER_SEC, &remote)) < 0) {
+        printf("sock_udp_recv() failed with %d\n", result);
+        goto out;
+    }
+
+    buffer[result] = 0;
+    printf("Received %d bytes - %s\n", result, buffer);
+
+out:
+    sock_udp_close(&sock);
+    return result;
+}
+
+SHELL_COMMAND(client, "Send UDP datagram", _client_cmd);
+
+void *server_thread(void *arg)
+{
+    (void)arg;
+    sock_udp_t sock;
+    sock_udp_ep_t local = { .family = AF_INET,
+                            .port = server_port };
+    sock_udp_ep_t remote;
+    int result;
+    int error;
+    char buffer[64];
+
+    int res = sock_udp_create(&sock, &local, NULL, 0);
+    if (res) {
+        printf("Sock_udp_create error!\n");
+        printf("Server stopped.\n");
+
+        mutex_lock(&server_mutex);
+        server_running = 0;
+        mutex_unlock(&server_mutex);
+
+        return NULL;
+    }
+
+    printf("Server started.\n");
+
+    while (1) {
+        if ((result = sock_udp_recv(&sock, buffer, sizeof(buffer) - 1, SOCK_NO_TIMEOUT, &remote)) < 0) {
+            printf("sock_udp_recv() failed with %d\n", result);
+            printf("Server stopped.\n");
+
+            sock_udp_close(&sock);
+
+            mutex_lock(&server_mutex);
+            server_running = 0;
+            mutex_unlock(&server_mutex);
+
+            return NULL;
+        }
+
+        buffer[result] = 0;
+        printf("Received %d bytes - %s\n", result, buffer);
+
+        error = sock_udp_send(&sock, buffer, result, &remote);
+        if (error < 0) {
+            printf("sock_udp_send() failed with %d\n", error);
+        }
+    }
+
+    sock_udp_close(&sock);
+
+    return NULL;
+}
+
+static int _server_cmd(int argc, char **argv)
+{
+    int is_running;
+    if (argc < 2) {
+        printf("usage: %s <port>\n", argv[0]);
+        return -1;
+    }
+
+    mutex_lock(&server_mutex);
+    is_running = server_running;
+    mutex_unlock(&server_mutex);
+
+    if(is_running == 1) {
+        printf("Server already running!\n");
+        return -1;
+    }
+
+    server_port = atoi(argv[1]);
+
+    mutex_lock(&server_mutex);
+    server_running = 1;
+    mutex_unlock(&server_mutex);
+
+    thread_create(server_stack, sizeof(server_stack), THREAD_PRIORITY_MAIN - 1, 0, server_thread, NULL, "server");
+
+    return -1;
+}
+
+SHELL_COMMAND(server, "Starts server which receives UDP datagrams", _server_cmd);
+
+#define _TEST_ADDR4_LOCAL  (0x0b64a8c0U)   /* 192.168.100.11 */
+#define _TEST_ADDR4_MASK   (0x0b64a8c0U)   /* 255.255.255.255 */
```

Comment:
Why not use `IPV4_ADDR_INIT()` and instead close my suggestion without a comment?

Reply:
Dear @benpicco ... I try to use this solution, but it cannot by used.
I provide a detailed explanation:
https://github.com/RIOT-OS/RIOT/pull/21519#discussion_r2598823865

Reply:
I also spotted today that I set wrong mask (rapid copy and paste :( ).
Fixed!